### PR TITLE
Navcurrent

### LINKED ISSUE #7643: Problem with the left menu. Version:  GoC.WebTemplate 4.0.21.2
The second level of the left menu is showing "Selected" for more than one file.

It should be associated only with the file "Documents.aspx" but it is also "selected" when displaying the file "DocumentsUpload.aspx".

Here is the code that I am using to generate the left menu in both pages:

``` vb
Dim oMenuSection As MenuSection = New MenuSection()

' Level 1
oMenuSection.Name = "Financial Transaction"
oMenuSection.Link = "FinancialTransaction.aspx"

' Level 2
oMenuSection.Items.Add(New MenuItem("ReviewersSignatories.aspx", "Reviewers and Signatories"))
oMenuSection.Items.Add(New MenuItem("Documents.aspx", "Supporting Documents"))
oMenuSection.Items.Add(New MenuItem("HistoricalVersions.aspx", "Historical Versions"))

' Add Menu Section
Page.WebTemplateMaster.WebTemplateCore.LeftMenuItems.Add(oMenuSection)
```

FYI, When removing the breadcrumbs in both pages, this problem disappear.

Thank you

file_1
file_2

### PR DESCRIPTION
**Pull request by ghost**
(DNM) Navcurrent Fix

### CODE REVIEW

**Comment by duboisp** [comment]
Have you seen the code comment i made on your original pr #7734? cheers

**Comment by kodecount** [comment]
@duboisp Sorry when I tried to squash my commits it destroyed the files so any comment is now gone.

**Comment by masterbee** [comment]
@duboisp do you mind taking a quick peek? it would be nice to get this P.R. in for this week's testing

**Comment by duboisp** [comment]
@masterbee I can but as this PR is from @kodecount and recently submitted by him I will prefer to have him to try to fix it first. 

@kodecount how long do you think it will take you to fix the code?

**Comment by kodecount** [comment]
@masterbee @duboisp I have reviewed the changes requested and well required some testing before updating these changes. I'll try to have it updated by mid next week.

**Comment by kodecount** [comment]
@masterbee Please have someone else finish fixing this I have no time to try to get GIT working right on my machine.

File: src/plugins/navcurrent/navcurrent.js [hunk 1]
```diff
@@ -74,14 +74,19 @@ var componentName = "wb-navcurr",
...
-							link = localBreadcrumbLinks[ i ];
-							linkHref = link.getAttribute( "href" );
-							if ( linkHref.length !== 0 && linkHref.charAt( 0 ) !== "#" ) {
-								localBreadcrumbLinksArray.push( link );
-								localBreadcrumbLinksUrlArray.push( link.hostname + link.pathname.replace( /^([^\/])/, "/$1" ) );
+						localBreadcrumbLinks = ( breadcrumb.jquery ? breadcrumb[ 0 ] : breadcrumb ).getElementsByTagName( "li" );
+						if(localBreadcrumbLinks) {
+							len = localBreadcrumbLinks.length;
+							if ( len && len > 0 ) {
+								link = localBreadcrumbLinks[ ( len - 1 ) ];
```
Comment (kodecount) [inline src/plugins/navcurrent/navcurrent.js]: From my research this is the only way to get the last element of an array in JavaScript so this line cannot change.

Sources:
http://www.codeblocq.com/2016/05/Get-the-last-element-of-an-Array-in-JavaScript/
http://www.w3resource.com/javascript-exercises/javascript-array-exercise-4.php
http://stackoverflow.com/questions/9050345/selecting-last-element-in-javascript-array

File: src/plugins/navcurrent/navcurrent.js [hunk 2]
```diff
@@ -74,14 +74,19 @@ var componentName = "wb-navcurr",
...
-							link = localBreadcrumbLinks[ i ];
-							linkHref = link.getAttribute( "href" );
-							if ( linkHref.length !== 0 && linkHref.charAt( 0 ) !== "#" ) {
-								localBreadcrumbLinksArray.push( link );
-								localBreadcrumbLinksUrlArray.push( link.hostname + link.pathname.replace( /^([^\/])/, "/$1" ) );
+						localBreadcrumbLinks = ( breadcrumb.jquery ? breadcrumb[ 0 ] : breadcrumb ).getElementsByTagName( "li" );
+						if( localBreadcrumbLinks ) {
+							len = localBreadcrumbLinks.length;
+							if ( len && len > 0 ) {
+								link = localBreadcrumbLinks[ ( len - 1 ) ];
```
Comment (kodecount) [inline src/plugins/navcurrent/navcurrent.js]: @duboisp From my research this is the only way to get the last element of an array in JavaScript so this line cannot change.

Sources:
http://www.codeblocq.com/2016/05/Get-the-last-element-of-an-Array-in-JavaScript/
http://www.w3resource.com/javascript-exercises/javascript-array-exercise-4.php
http://stackoverflow.com/questions/9050345/selecting-last-element-in-javascript-array

Comment (kodecount) [inline src/plugins/navcurrent/navcurrent.js]: To change `localBreadcrumbLinks` so it not array would require an entire change of the navcurrent plugin.

File: src/plugins/navcurrent/navcurrent.js [hunk 3]
```diff
@@ -74,14 +74,19 @@ var componentName = "wb-navcurr",
...
-							link = localBreadcrumbLinks[ i ];
-							linkHref = link.getAttribute( "href" );
-							if ( linkHref.length !== 0 && linkHref.charAt( 0 ) !== "#" ) {
-								localBreadcrumbLinksArray.push( link );
-								localBreadcrumbLinksUrlArray.push( link.hostname + link.pathname.replace( /^([^\/])/, "/$1" ) );
+						localBreadcrumbLinks = ( breadcrumb.jquery ? breadcrumb[ 0 ] : breadcrumb ).getElementsByTagName( "li" );
+						if ( localBreadcrumbLinks ) {
+							len = localBreadcrumbLinks.length;
+							if ( len && len > 0 ) {
+								link = localBreadcrumbLinks[ ( len - 1 ) ];
```
Comment (kodecount) [inline src/plugins/navcurrent/navcurrent.js]: @duboisp From my research this is the only way to get the last element of an array in JavaScript so this line cannot change.

Sources:
http://www.codeblocq.com/2016/05/Get-the-last-element-of-an-Array-in-JavaScript/
http://www.w3resource.com/javascript-exercises/javascript-array-exercise-4.php
http://stackoverflow.com/questions/9050345/selecting-last-element-in-javascript-array

To change `localBreadcrumbLinks` so it not array would require an entire change of the navcurrent plugin.

**Comment by kodecount** [comment]
@duboisp @masterbee Got my git problem fixed I had node 7 and it was causing issue.

**Comment by kodecount** [comment]
@duboisp @LaurentGoderre @masterbee Can you please review what making this fail? It seems to be a issue with travis?

**Comment by LaurentGoderre** [comment]
Restarted the build phantomjs had a random blip

**Comment by kodecount** [comment]
@LaurentGoderre Whatever that means.

**Comment by kodecount** [comment]
@LaurentGoderre I think I broke phantomjs (woot woot).

**Comment by LaurentGoderre** [comment]
Usually, thus is caused by connectivity issues.

**Comment by kodecount** [comment]
I'm going to try to create a new PR.

**Comment by LaurentGoderre** [comment]
It wouldn't change anything. Better just wait until tomorrow and try again

**Pull request closed without merging**